lib/matplotlib/backends/backend_pdf.py

```diff
@@ -1711,22 +1709,13 @@ def check_gc(self, gc, fillcolor=None):
         gc._fillcolor = orig_fill
         gc._effective_alphas = orig_alphas
 
-    def track_characters(self, font, s):
-        """Keeps track of which characters are required from each font."""
-        if isinstance(font, str):
-            fname = font
-        else:
-            fname = font.fname
-        realpath, stat_key = cbook.get_realpath_and_stat(fname)
-        used_characters = self.file.used_characters.setdefault(
-            stat_key, (realpath, set()))
-        used_characters[1].update(map(ord, s))
-
-    def merge_used_characters(self, other):
-        for stat_key, (realpath, charset) in other.items():
-            used_characters = self.file.used_characters.setdefault(
-                stat_key, (realpath, set()))
-            used_characters[1].update(charset)
+    @cbook.deprecated("3.3", alternative="renderer.file.track_characters")
```

Comment:
Is this a method users need? If so, I would consider not deprecating it. How/where the tracking is handled feels like an implementation detail. Users shouldn't know about this and they shouldn't drop to accessing lower-level objects to perform the action.

If this is not user-facing, just deprecate without alternative.

Reply:
removed alternatives